lib/ansible/module_utils/six/__init__.py

```diff
@@ -676,6 +677,8 @@ def u(s):
         _assertRaisesRegex = "assertRaisesRegex"
         _assertRegex = "assertRegex"
         _assertNotRegex = "assertNotRegex"
+    utcfromtimestamp = functools.partial(datetime.datetime.fromtimestamp, tz=datetime.timezone.utc)
+    utcnow = functools.partial(datetime.datetime.now, tz=datetime.timezone.utc)
```

Comment:
@mattclay Why do the functions need to return naive datetime objects on Python 3? All of the things using these module_util functions are now comparing offset aware datetime objects on Python 3, and none of the modules return the datetime string, just use these objects to calculate the elapsed seconds, which should have the same result as using offset naive. I may be overlooking something though.

And yes, I will move them to `module_utils.compat.datetime` :+1:

Reply:
@s-hertel If these functions were in modules, there would be no issue. However, since they're in module_utils, we should assume they'll be used in collections at some point. As a public API, they should return either naive or aware datetime objects, regardless of the Python version used.

If we're going to be returning naive datetime objects, perhaps it's better these functions only exist in modules, so we don't encourage their usage. If they return aware datetime objects, there's some advantage to having them available in module_utils, since we'd also need to implement a `UTC` `tzinfo`, providing compatibility with Python 2.7.

Reply:
Updated to aware datetime objects for Python 2.7. I could also rename the functions to make this more obvious, maybe `utcnow_aware` and `utcfromtimestamp_aware`?